save-frontend/README.md

```diff
@@ -7,4 +7,149 @@ To run frontend locally, use `./gradlew :save-frontend:browserDevelopmentRun --c
+5. `add_header Last-Modified "";`
+   The `Last-Modified` header is used for caching validation, but setting it to an empty value indicates that the server does not want to participate in any caching validation based on the last modification date of the resource.
+6. `if_modified_since off;`
+   The `if_modified_since` directive controls the behavior of the `If-Modified-Since` request header.
+   Disabling it ensures that clients won't use conditional requests, which can be helpful in certain caching scenarios or when caching behavior needs to be controlled explicitly.
+
+### `webpack-dev-server` configuration for no `api-gateway` run
+Here is a `webpack-dev-server` configuration for running without `api-gateway` on:
+```javascript
+config.devServer = Object.assign(
+  {},
+  config.devServer || {},
+  {
+    setupMiddlewares: (middlewares, devServer) => {
+      devServer.app.get("/sec/user", (req, res) => { return res.send("admin"); });
+      devServer.app.get("/sec/oauth-providers", (req, res) => { return res.send([]); });
+      return middlewares;
+    },
+    proxy: [
+      {
+        context: ["/api/sandbox/**"],
+        target: 'http://localhost:5400',
+        logLevel: 'debug',
+        onProxyReq: function (proxyReq, req, res) {
+          proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+          proxyReq.setHeader("X-Authorization-Source", "basic");
+        }
+      },
+      {
+        context: ["/api/demo/**"],
+        target: 'http://localhost:5421',
+        logLevel: 'debug',
+        onProxyReq: function (proxyReq, req, res) {
+          proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+          proxyReq.setHeader("X-Authorization-Source", "basic");
+        }
+      },
+      {
+        context: ["/api/cpg/**"],
+        target: 'http://localhost:5500',
+        logLevel: 'debug',
+        onProxyReq: function (proxyReq, req, res) {
+          proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+          proxyReq.setHeader("X-Authorization-Source", "basic");
+        }
+      },
+      {
+        context: ["/api/**"],
+        target: 'http://localhost:5800',
+        logLevel: 'debug',
+        onProxyReq: function (proxyReq, req, res) {
+          proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+          proxyReq.setHeader("X-Authorization-Source", "basic");
+        }
+      }
+    ],
+    historyApiFallback: true
+  }
+);
+```
+
+`setupMiddlewares` sets stubs for `/sec/user` and `/sec/oauth-providers` endpoints.
+
+`historyApiFallback` makes `webpack-dev-server` return `index.html` in case of `404 Not Found`.
+
+Sometimes it is useful to add authorization headers when proxying to some Spring services e.g. `save-backend`.
+It can be done with setting `onPorxyReq`:
+```javascript
+onProxyReq: (proxyReq, req, res) => {
+  proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+  proxyReq.setHeader("X-Authorization-Source", "basic");
```

Comment:
it's not used anymore (already in master)

Reply:
What do you mean? Did someone merge the config for `gateway` local run or what?

Reply:
Or was it you who changed the behavior?

Reply:
If you changed it, update the docs please 🥺🥺

Reply:
`X-Authorization-Source` was removed

> If you changed it, update the docs please 🥺🥺

Will do

Reply:
will do in scope https://github.com/saveourtool/save-cloud/issues/2407